### DISCUSSION TITLE
PINNs for larger length and time domains

### DISCUSSION THREAD

**Discussion by ryabs07** [lululxvi/deepxde#1487] [Q&A]
Hi all,
Has anyone experience with implementing PINNs on larger length and time domains (for instance, L>1000m and T>20,000 secs)? I have tried non-dimensionalizing/scaling the PDEs. This was working for domains in the order of  L=10^2 and T=10^2. But it doesnt seem to work on domains larger than that. If anyone has suggestions on tackling this kind of problem, I would be very grateful. :)
Thanks in advance.

**Comment by lululxvi** [comment]
Non-dimensionalize the domain to [0, 1].

**Reply by Mouhamedsaw** [reply]
Thank you in advance Lu Lu and ryabs07,

I am dealing with a larger time domain problem as described above (T>20,000 secs).
When I set the final time Tf = 20,000 secs in the code, the results are not correct and the visualisation at the initial condition is not good.
And when I set the final time Tf = 1 sec, the visualisation at the initial condition is fine. But to see the physical process, I need a larger time (T>20,000 secs).

My question is if I need to add a line of code in the deepxde implementation if I do not dimension my domain [0, 20,000].

Thank you for your time and patience.

**Reply by ryabs07** [reply]
Hi,
Unfortunately i haven't been able to solve the problem yet. The non dimensionalization of domain didnt work either. 
As per my literature study, there are few variations of PINNs, such as domain decomposition method and Finite Basis PINNs, that can deal with this problem. 
I couldn't test these variations in my study as they were out of the scope of my Master's Thesis. But if you have any progress on this problem, do let me know. 

Best wishes.

**Reply by Mouhamedsaw** [reply]
Thank you for your reply,

Yes, I will have a look at these variations of PINNs.

Sure, I will get back to you.

Best wishes.
